docs/assessment.md

```diff
@@ -226,42 +226,60 @@ Students are able to see their groupmates' UIDs, which can become a point of con
 
-To enable these features, set `advanceScorePerc` to any number between 0 and 100 at the `assessment`, `zone`, `alternative group`, or `question` level. An example of what this looks like is below, with boilerplate attributes omitted:
+To enable these features, set `advanceScorePerc` for any question to a number between 0 and 100. An example of what this looks like is below, with boilerplate attributes omitted:
 
 ```json
 {
-  "advanceScorePerc": 100,
   "zones": [
     {
-      "advanceScorePerc": 80,
       "questions": [
-        {
-          "id": "page1",
-          "advanceScorePerc": 50
-        },
-        {
-          "id": "page2"
-        },
-        {
-          "id": "page3"
-        }
+        { "id": "q1", "advanceScorePerc": 50 },
+        { "id": "q2", "advanceScorePerc": 100 },
+        { "id": "q3", "advanceScorePerc": 0 },
+        { "id": "q4", "advanceScorePerc": 80 },
+        { "id": "q5", "advanceScorePerc": 100 }
       ]
     }
   ]
 }
 ```
 
-In the above example, a student will need to score at least 50% on `page1` in order to unlock `page2`. Since `page2` has no `advanceScorePerc` set at the question-level, it looks for the next-closest level in the tree where it is defined, which turns out to be the zone level. Thus, `page2` requires a score of at least 80% in order to unlock `page3`. Because `advanceScorePerc` is defined at the zone-level for all questions, the value 100 at the assessment level is never used to determine the minimum advancement score for any question.
+Each question blocks all later questions until its `advanceScorePerc` is met. In the above example, `q1` blocks all later questions until the student has scored at least 50% on it. Then `q2` blocks all later questions until the student has a perfect score on it. Once a student gets past `q2`, both `q3` and `q4` are immediately available because `q3` does not do any blocking. Finally, `q4` blocks the remaining `q5` until the student has an 80% score on it. The `advanceScorePerc` attribute on `q5` is irrelevant because there are no questions after it.
 
-If a student uses all of their attempts on a question and cannot submit any more attempts, the next question will automatically unlock, no matter what score they earned on the previous question. This is to prevent students from getting permanently stuck on an assessment, unable to receive further credit.
+The relevant score for comparing to `advanceScorePerc` is the student's _highest submission score_ for the question, not their percentage score on the question overall. For example, suppose `q1` above has `"points": [10, 4, 2, 1]`. Then a student who makes a 50%-correct submission on their second attempt will unblock the question, even though they only score 2 points out of 10 on the question (50% of the 4-point second-chance value).
```

Comment:
For auto points or combined (auto+manual)?

Reply:
Good question. Here's the code: https://github.com/PrairieLearn/PrairieLearn/blob/5b65573e6974251dbbcef171f68013a33174d3ef/apps/prairielearn/src/sprocs/question_order.sql#L22
Does `highest_submission_score` include manual grading? Do we want it to for this use case?

Reply:
I checked the source, and it seems it's not consistent. From what I could tell, it is the auto score before any manual grading is done, and total score after it. This should probably reviewed to make it more consistent.

Reply:
Sorry, I read it wrong. It's auto score, with 0 if the internal/external grader never ran.

Reply:
In essence what this means is that the advance score percentage applies to auto points (i.e., internal or external grader). So this option should not be used for questions that are manual-only. This is probably an important note to add here, or something to change in the implementation (in the sense of ignoring this option if the question has manual-only points).

Reply:
I added a note here: https://github.com/PrairieLearn/PrairieLearn/blob/442ca8a6220377b4c38d6c48dd91e48142859398/docs/assessment.md?plain=1#L253

I agree that it would probably be good to enforce that purely-manually-graded questions don't have `advanceScorePerc` set, or issue a warning, or something.